common/util/file_util.cc

```diff
@@ -161,6 +167,62 @@ absl::Status GetContents(absl::string_view filename, std::string *content) {
   return absl::OkStatus();
 }
 
+absl::StatusOr<std::unique_ptr<MemBlock>> AttemptMemMapFile(
+    absl::string_view filename) {
+#ifndef _WIN32
+  class MemMapBlock final : public MemBlock {
+   public:
+    MemMapBlock(char *buffer, size_t size) : buffer_(buffer), size_(size) {}
+    ~MemMapBlock() final { munmap(buffer_, size_); }
+    absl::string_view AsStringView() const final { return {buffer_, size_}; }
+
+   private:
+    char *const buffer_;
+    const size_t size_;
+  };
+
+  const std::string nul_terminated_filename(filename);
+  const int fd = open(nul_terminated_filename.c_str(), O_RDONLY);
+  if (fd < 0) {
+    return CreateErrorStatusFromErrno("Can't open file");
+  }
+
+  struct stat s;
+  if (fstat(fd, &s) < 0) {
+    close(fd);
+    return CreateErrorStatusFromErrno("can't stat");
+  }
+
+  const size_t file_size = s.st_size;
+  void *const buffer = mmap(NULL, file_size, PROT_READ, MAP_SHARED, fd, 0);
+  close(fd);
+  if (buffer == MAP_FAILED) {
+    return CreateErrorStatusFromErrno("Can't mmap file");
+  }
+
+  return std::make_unique<MemMapBlock>((char *)buffer, file_size);
+#else
+  // TODO: implement some memory mapping on Windows.
+  return absl::UnimplementedError("No windows mmap implementation yet.");
+#endif
+}
+
+absl::StatusOr<std::unique_ptr<MemBlock>> GetContentAsMemBlock(
+    absl::string_view filename) {
+  auto mmap_result = AttemptMemMapFile(filename);
+  if (mmap_result.status().ok()) {
+    return mmap_result;
+  }
+
```

Comment:
log info, or warning here

Reply:
There are a few valid reasons tht this wouldn't work, e.g. reading from a pipe of sorts, so it is an expected failure to not be able to `mmap()` something. Logging that would be confusing.